src/entrypoint/trigger.ts

```diff
@@ -77,12 +82,29 @@ export async function onchainTransactionEventHandler(tx) {
       { success: true, pending: false, transactionType: "payment" },
       "payment completed",
     )
-    const entry = await Transaction.findOne({
-      account_path: "Liabilities",
+
+    const entries = await Transaction.find({
+      account_path: liabilitiesMainAccount,
```

Comment:
why not adding something like:
`accounts: { $ne: bankOwnerPath }`
in the query?